src/modules/paymentForm.ts

```diff
@@ -0,0 +1,198 @@
+                <input type="text" class="input input_address" placeholder="Delivery address">
+                <input type="email" class="input input_email" placeholder="E-mail">
+                <input type="text" class="input input_valid" id="input_valid" placeholder="Validate date mm/yy">
+                <input type="text" class="input input_code" id="input_code" placeholder="CVV">
+            </div>
+            <div class="buttons__wrapper">
+            <button type="submit" class="button button-pay" disabled>Pay</button>
+            </div>
+        </form>`;
+    }
+
+    public drawForm(): void {
+        this.parent.appendChild(this.modalWindow);
+
+        const form = document.querySelector('.modal__form')!;
+        form.addEventListener('input', () => this.validateForm());
+
+        const cardNumber = document.querySelector(
+            '.input_card-number'
+        )! as HTMLInputElement;
+        cardNumber.addEventListener('input', () => this.updateCardImage());
+
+        const close: Element = document.querySelector('.modal__close')!;
+        close.addEventListener('click', () => this.closeWindow());
+
+        const buttonPay: Element = document.querySelector('.button-pay')!;
+        buttonPay.addEventListener('click', () => this.onClickButton('pay'));
+    }
+
+    private closeWindow(): void {
+        this.modalWindow.remove();
+    }
+
+    private checkCardNumber(): boolean {
+        const cardNumber = document.querySelector(
+            '.input_card-number'
+        )! as HTMLInputElement;
+        cardNumber.maxLength = MAX_LENGTH_CARD_NUMBER;
+        const value = cardNumber.value;
+        cardNumber.value = value
+            .replace(/[^\d]/g, '')
+            .replace(/(.{4})/g, '$1 ')
+            .trim();
+
+        if (value.length === MAX_LENGTH_CARD_NUMBER) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private updateCardImage(): void {
+        const cardNumber = document.querySelector(
+            '.input_card-number'
+        )! as HTMLInputElement;
+        const cardImage = document.querySelector(
+            '.img_payments'
+        )! as HTMLImageElement;
+        const value = cardNumber.value;
+        if (value.charAt(0) === '4') {
+            cardImage.src = '../../assets/icons/visa.png';
+        } else if (value.charAt(0) === '5') {
+            cardImage.src = '../../assets/icons/mastercard.png';
+        } else {
+            cardImage.src = '../../assets/icons/maestro.png';
+        }
+    }
+
+    private checkName(): boolean {
+        const fullName = document.querySelector(
+            '.input_name'
+        )! as HTMLInputElement;
+        const value = fullName.value;
+        const regexp: RegExp = /^[A-Za-z]{3,}\s[A-Za-z]{3,}(\s?[A-Za-z]{3,})*$/;
+        if (regexp.test(value)) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
????
Cab we just return regexp.test(value)?

Reply:
Fixed